### ISSUE TITLE
assembleRelease build failing due to heap integration

### ISSUE THREAD

**Issue by anshulporwal26** [heap/react-native-heap#246]
Prerequisites:
react-native 0.63.3
React Navigation 4.3.2
heap version 0.15.0

While building the app using `./gradlew assembleRelease`, getting the following error and build is failing. Please suggest a fix for this issue. Thanks

<img width="1068" alt="Screenshot 2021-05-06 at 4 51 12 PM" src="https://user-images.githubusercontent.com/16168742/117290323-4a968e00-ae8b-11eb-9d4e-850dd2e45100.png">

**Comment by ShotSkydiver** [comment]
YEP I have spent the last few days struggling with failing CI builds and just nonstop poring over every aspect of my mobile code and build settings with bitrise and debugging and thinking it was one issue or another and it wasn't until an hour ago that it dawned on me that it was another dev's recent integration of this very package and YEP there's something really broken with this current heap package version and babel traverse, this is like.... a critical issue, whoever's maintaining this repo should prioritize this issue, it's absolutely a dealbreaker

**Comment by matt-oakes** [comment]
We also have the same issue whenever bundling is happening using these versions:

react-native 0.64.0
React Navigation 5
Heap version 0.15.0

**Comment by cristianbote** [comment]
This fails on `ios` as well. The only version that _works_ is 0.12.0. 😬

**Comment by matt-oakes** [comment]
Related issue, but something in this library is triggering it https://github.com/babel/babel/issues/8028

**Comment by matt-oakes** [comment]
As a workaround for this issue, we are using [`patch-package`](https://github.com/ds300/patch-package) and applying this patch directly to `babel-transform` which is where the issue is occurring:

```
diff --git a/node_modules/babel-traverse/lib/path/evaluation.js b/node_modules/babel-traverse/lib/path/evaluation.js
index 86aac21..3d1a2e9 100644
--- a/node_modules/babel-traverse/lib/path/evaluation.js
+++ b/node_modules/babel-traverse/lib/path/evaluation.js
@@ -156,7 +156,7 @@ function evaluate() {
         return deopt(binding.path);
       }
 
-      if (binding && path.node.start < binding.path.node.end) {
+      if (binding && binding.path && binding.path.node && path.node.start < binding.path.node.end) {
         return deopt(binding.path);
       }
 
```

**Comment by ShotSkydiver** [comment]
thanks for leading me in the right direction with your patch-package fix! here's my patch that's simply updated to work with the latest version of the babel traverse package, v7.14, the naming convention is different (@babel/traverse instead of babel-traverse) and the actual code changes in the patch differ as well. Hopefully this will help others!

```
diff --git a/node_modules/@babel/traverse/lib/path/evaluation.js b/node_modules/@babel/traverse/lib/path/evaluation.js
index 1bea680..95f6706 100644
--- a/node_modules/@babel/traverse/lib/path/evaluation.js
+++ b/node_modules/@babel/traverse/lib/path/evaluation.js
@@ -125,7 +125,7 @@ function _evaluate(path, state) {
       return deopt(binding.path, state);
     }

-    if (binding && path.node.start < binding.path.node.end) {
+    if (binding && binding.path && binding.path.node && path.node.start < binding.path.node.end) {
       return deopt(binding.path, state);
     }
```

**Comment by joeyfigaro** [comment]
Error still occurring on at least iOS. Running into same error when I add `'./node_modules/@heap/react-native-heap/instrumentor/src/index.js'` to my babel plugins. The error I'm seeing isn't for TextInput, however–it's for TouchableOpacity.

**Comment by bnickel** [comment]
A fix for this was implemented in #256 and shipped as part of 0.17.0.  Please reopen if this wasn't resolved.
